addons/im_livechat_mail_bot/controllers/thread.py

```diff
@@ -0,0 +1,12 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.http import request, route
+from odoo.addons.mail.controllers import thread
+
+
+class ThreadController(thread.ThreadController):
+    @route()
+    def mail_message_post(self, thread_model, thread_id, post_data, context=None):
+        if post_data.get("canned_response_ids"):
```

Comment:
now that we have canned response in mail, shouldn't this be in mail code?
or at least just `mail_bot`?

Reply:
I don't remember what we discussed, but wasn't it possible to move the check of mail bot somewhere else, where `canned_response_ids` is accessible? In override of route?

Reply:
We had this discussion with PO after confirming [this task](https://www.odoo.com/web#id=3570249&cids=1&menu_id=6478&action=4043&model=project.task&view_type=form) that canned responses could be triggered without livechat module in chat with OdooBot. I suggested we do it in a [separate task](https://www.odoo.com/web#id=3848027&cids=1&menu_id=6478&action=4043&model=project.task&view_type=form) to delete livechat module (needs an upgrade) which is ready and waiting for this PR to be merged. So it will be moved to the mail_bot module, when livechat_mail_bot is removed.